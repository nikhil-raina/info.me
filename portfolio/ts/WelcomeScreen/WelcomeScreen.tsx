/**
 * @providesModule WelcomeScreen
 * @flow
 */

import React from 'react';
import styles from './WelcomeScreenStyle';
import {View, Text, Button} from 'react-native';

export interface Props {
  name: string;
  enthusiasmLevel?: number;
}

interface State {
  enthusiasmLevel: number;
}

export default class WelcomeScreen extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    if ((props.enthusiasmLevel || 0) <= 0) {
      throw new Error('You could be a little more enthusiastic. :D');
    }

    this.state = {
      enthusiasmLevel: props.enthusiasmLevel || 1,
    };
  }

  onIncrement = () => {
    this.setState({
      enthusiasmLevel: this.state.enthusiasmLevel + 1,
    });
  };

  onDecrement = () => {
    this.setState({
      enthusiasmLevel: this.state.enthusiasmLevel - 1,
    });
  };

  getExclamationMarks = (numChars: number) => Array(numChars + 1).join('!');

  render(): React.ReactElement {
    return (
      <View style={styles.root}>
        <Text style={styles.greeting}>
          Hello{' '}
          {this.props.name +
            this.getExclamationMarks(this.state.enthusiasmLevel)}
        </Text>

        <View style={styles.buttons}>
          <View style={styles.button}>
            <Button
              title="-"
              onPress={this.onDecrement}
              accessibilityLabel="decrement"
              color="red"
            />
          </View>

          <View style={styles.button}>
            <Button
              title="+"
              onPress={this.onIncrement}
              accessibilityLabel="increment"
              color="blue"
            />
          </View>
        </View>
      </View>
    );
  }
}
