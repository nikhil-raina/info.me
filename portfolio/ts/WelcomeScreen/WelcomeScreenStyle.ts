/**
 * @provideModule WelcomeScreenStyle
 * @flow
 */

import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#F2E6C2',
    alignItems: 'center',
    justifyContent: 'center',
  },
  textStyleHeading: {
    fontSize: 40,
    color: '#002942',
  },
  textStyleSubHeading: {
    paddingTop: 5,
    fontSize: 25,
    color: '#5F6B6E',
    padding: 30,
  },
});
