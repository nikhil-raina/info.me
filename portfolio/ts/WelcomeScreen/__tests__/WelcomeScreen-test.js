import React from 'react';
import renderer from 'react-test-renderer';

import WelcomeScreen from '../WelcomeScreen';

describe('Rendering basic demo screen', () => {
  it('renders correctly', () => {
    const screen = renderer.create(
      <WelcomeScreen name={'World'} enthusiasmLevel={1} />,
    );

    expect(screen.toJSON()).toMatchSnapshot();
  });
});
