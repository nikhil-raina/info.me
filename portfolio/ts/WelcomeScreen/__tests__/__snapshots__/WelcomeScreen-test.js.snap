// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering basic demo screen renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F2E6C2",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#002942",
        "fontSize": 40,
      }
    }
  >
    Hello
  </Text>
  <Text
    style={
      Object {
        "color": "#5F6B6E",
        "fontSize": 25,
        "padding": 30,
        "paddingTop": 5,
      }
    }
  >
    Hello this is the subtitle text that I will be putting in here
  </Text>
</View>
`;
